addon/scripts/tokens.js

```diff
@@ -23,16 +23,27 @@
 //  r blinding factor, sjcl bignum
 function CreateBlindToken() {
     let t = newRandomPoint();
+    if (!t.point) {
+        return null;
+    }
     let bpt = blindPoint(t.point);
     return { token: t.token, point: bpt.point, blind: bpt.blind };
 }
 
 // returns: array of blind tokens
 function GenerateNewTokens(n) {
+    let tokens = [];
     let i = 0;
-    let tokens = new Array(n);
-    for (i = 0; i < tokens.length; i++) {
-        tokens[i] = CreateBlindToken();
+    while (i < n) {
+        let tok = CreateBlindToken();
+        if (!tok) {
+            console.warn("[privacy-pass]: Tried to generate a random point on the curve, but failed.");
+            console.warn("[privacy-pass]: Will drop the null point.");
+            n--;
```

Comment:
this looks weird (as far as my javascript knowledge goes). as n is passed by copy and is not read anywhere in this function, I think reducing it has no effect. Especially that n is a lvalue.
see https://js.do/code/axrev1

Reply:
(see above)